azurerm/internal/services/postgres/postgresql_flexible_server_configuration.go

```diff
@@ -0,0 +1,172 @@
+			Value:  utils.String(d.Get("value").(string)),
+			Source: utils.String("user-override"),
+		},
+	}
+
+	future, err := client.Update(ctx, resGroup, serverName, name, props)
+	if err != nil {
+		return err
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, resGroup, serverName, name)
+	if err != nil {
+		return err
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Azure Postgresql Flexible Server configuration %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceFlexibleServerConfigurationRead(d, meta)
+}
+
+func resourceFlexibleServerConfigurationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Postgres.FlexibleServersConfigurationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FlexibleServerConfigurationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.FlexibleServerName, id.ConfigurationName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] Azure Postgresql Flexible Server configuration '%s' was not found (resource group '%s')", id.ConfigurationName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure Postgresql Flexible Server configuration %s: %+v", id.ConfigurationName, err)
+	}
+
+	d.Set("name", id.ConfigurationName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("server_name", id.FlexibleServerName)
+
+	if props := resp.ConfigurationProperties; props != nil {
+		d.Set("value", props.Value)
+	}
+
+	return nil
+}
+
+func resourceFlexibleServerConfigurationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Postgres.FlexibleServersConfigurationsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FlexibleServerConfigurationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.FlexibleServerName, id.ConfigurationName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Azure Postgresql Flexible Server configuration '%s': %+v", id.ConfigurationName, err)
+	}
+
+	props := postgresqlflexibleservers.Configuration{
+		ConfigurationProperties: &postgresqlflexibleservers.ConfigurationProperties{
+			Value:  resp.DefaultValue,
+			Source: utils.String("user-override"),
+		},
+	}
+
+	future, err := client.Update(ctx, id.ResourceGroup, id.FlexibleServerName, id.ConfigurationName, props)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return err
```

Comment:
We should wrap the error response here

```suggestion
		return fmt.Errorf("waiting for deletion of %q: %+v", id, err)
```

Reply:
Improved error messages